visualization/dashboard/src/MLIDashboard/NewExplanationDashboard.tsx

```diff
@@ -390,19 +433,38 @@ export class NewExplanationDashboard extends React.PureComponent<IExplanationDas
         this.setState({sortVector: ModelExplanationUtils.getSortIndices(this.state.cohorts[0].calculateAverageImportance()).reverse()});
     }
 
-    private onCohortChange(newCohort: Cohort, index: number): void {
+    private onCohortChange(newCohort: Cohort): void {
         const prevCohorts = [...this.state.cohorts];
-        prevCohorts[index] = newCohort;
-        this.setState({cohorts: prevCohorts});
+        prevCohorts[this.state.editingCohortIndex] = newCohort;
+        this.setState({cohorts: prevCohorts, editingCohortIndex: undefined});
     }
 
-    private deleteCohort(index: number): void {
+    private deleteCohort(): void {
         const prevCohorts = [...this.state.cohorts];
-        prevCohorts.splice(index, 1);
+        prevCohorts.splice(this.state.editingCohortIndex, 1);
         this.setState({cohorts: prevCohorts});
     }
 
     private clearWarning(): void {
         this.setState({validationWarnings: []})
     }
+
+    private clearSizeWarning(): void {
```

Comment:
Why separate the size warning from other validation warnings?

Reply:
Right now there are two warning toasts, one for validation, another warning that size can impact performance. If we need a third, I'll abstract it to a list rather than n specific warnings